godot-core/src/builtin/signal.rs

```diff
@@ -0,0 +1,203 @@
+
+        Error::from_godot(error as i32)
+    }
+
+    /// Disconnects this signal from the specified [`Callable`].
+    ///
+    /// If the connection does not exist, generates an error. Use [`Self::is_connected`] to make sure that the connection exists.
+    pub fn disconnect(&self, callable: Callable) {
+        self.as_inner().disconnect(callable);
+    }
+
+    /// Emits this signal.
+    ///
+    /// All Callables connected to this signal will be triggered.
+    pub fn emit(&self, varargs: &[Variant]) {
+        let Some(mut object) = self.object() else {
+            return;
+        };
+
+        object.emit_signal(self.name(), varargs);
+    }
+
+    /// Returns an [`Array`] of connections for this signal.
+    ///
+    /// Each connection is represented as a Dictionary that contains three entries:
+    ///  - `signal` is a reference to this [`Signal`];
+    ///  - `callable` is a reference to the connected [`Callable`];
+    ///  - `flags` is a combination of [`ConnectFlags`](crate::engine::object::ConnectFlags).
+    ///
+    /// _Godot equivalent: `get_connections`_
+    pub fn connections(&self) -> Array<Dictionary> {
+        self.as_inner()
+            .get_connections()
+            .iter_shared()
+            .map(|variant| variant.to())
+            .collect()
+    }
+
+    /// Returns the name of the signal.
+    pub fn name(&self) -> StringName {
+        self.as_inner().get_name()
+    }
+
+    /// Returns the object to which this signal belongs.
+    ///
+    /// Returns [`None`] when this signal doesn't have any object.
+    ///
+    /// _Godot equivalent: `get_object`_
+    pub fn object(&self) -> Option<Gd<Object>> {
+        self.as_inner().get_object().map(|mut object| {
+            <Object as Bounds>::DynMemory::maybe_inc_ref(&mut object.raw);
+            object
+        })
+    }
+
+    /// Returns the ID of this signal's object, see also [`Gd::instance_id`].
+    ///
+    /// Returns [`None`] when this signal doesn't have any object.
+    ///
+    /// _Godot equivalent: `get_object_id`_
+    pub fn object_id(&self) -> Option<InstanceId> {
+        let id = self.as_inner().get_object_id();
+        InstanceId::try_from_i64(id)
+    }
+
+    /// Returns `true` if the specified [`Callable`] is connected to this signal.
+    pub fn is_connected(&self, callable: Callable) -> bool {
+        self.as_inner().is_connected(callable)
+    }
+
+    /// Returns `true` if the signal's name does not exist in its object, or the object is not valid.
+    pub fn is_null(&self) -> bool {
+        self.as_inner().is_null()
+    }
+
+    /// Creates an invalid/empty object that is not able to be called.
+    ///
+    /// _Godot equivalent: `Signal()`_
+    pub fn invalid() -> Self {
+        unsafe {
+            Self::from_sys_init(|self_ptr| {
+                let ctor = sys::builtin_fn!(signal_construct_default);
+                ctor(self_ptr, ptr::null_mut())
+            })
+        }
+    }
```

Comment:
I would list constructors at the beginning of the `impl` block.

Also, should be "signal", not "object" in the doc.